astropy/utils/data.py

```diff
@@ -1003,6 +1005,14 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None):
             with shelve.open(urlmapfn) as url2hash:
                 if url_key in url2hash:
                     return url2hash[url_key]
+                # If there is a cached copy from mirror, use it.
+                else:
+                    for cur_url in (conf.dataurl, conf.dataurl_alias):
```

Comment:
Note: I am open to a more elegant way to do this.

Reply:
You can decide if it's really "elegant", but this at least has the advantage of not requiring that  `dataurl_alias` be specified:

```
dataurls_to_alias = {}  # this could probably just sit at module-level global cope?

# do this whenever the first time is the dataurl is actually *accessed*
from urllib.requests import urlopen

u = urlopen(conf.dataurl)
u.close()
dataurls_to_alias[conf.dataurl] = u.geturl()

# then at the line this comment is at, do this:
urls_to_try = [conf.dataurl]
if conf.dataurl in dataurls_to_alias:
    urls_to_try.append(dataurls_to_alias[conf.dataurl])
for cur_url in urls_to_try:
    ...

```

Reply:
Okay, I'll give it a try. Thank you!

Reply:
Looking at this more, this solution will be harder to backport. I'll implement Python 3 only code in this PR and comment on how to make it Python 2 compatible.